custom/_legacy/pact/api.py

```diff
@@ -38,37 +39,8 @@
 FORM_ADDRESS = "Address and Phone Update"
 
 
-def get_cloudcare_app():
-    """
-    Total hack function to get direct links to the cloud care application pages
-    """
-
-    from corehq.apps.cloudcare import api
-
-    app = api.get_cloudcare_app(PACT_DOMAIN, PACT_CLOUD_APPNAME)
-    app_id = app['_id']
-
-    pact_cloudcare = filter(lambda x: x['name']['en'] == PACT_CLOUDCARE_MODULE, app['modules'])
-    forms = pact_cloudcare[0]['forms']
-    ret = dict((f['name']['en'], ix) for (ix, f) in enumerate(forms))
-
-    url_root = '/a/%(domain)s/cloudcare/apps/view/%(build_id)s/%(module_id)s/%(form_id)s/case/%(case_id)s/enter/'
-    ret['url_root'] = url_root
-    ret['domain'] = PACT_DOMAIN
-    ret['app_id'] = app_id
-    latest_build_id = get_latest_build_id(PACT_DOMAIN, app_id)
-    if latest_build_id is not None:
-        ret['build_id'] = latest_build_id
-    ret['module_id'] = 0
-    return ret
-
-
 def get_cloudcare_url(case_id, mode):
-    app_dict = get_cloudcare_app()
-    url_root = app_dict['url_root']
-    build_id = app_dict['build_id']
-    module_id = app_dict['module_id']
-    return  url_root % dict(form_id=app_dict[mode], case_id=case_id, domain=PACT_DOMAIN, build_id=build_id, module_id=module_id)
+    return reverse('cloudcare_default', args=[PACT_DOMAIN])
```

Comment:
You could probably add `#` plus url-encoded/quoted `{"appId": "1309b791a8ca4f36be6d2df0212da0b3", "page": null, "search": null, "steps": [0, 1, "4f997b6974fb20ab715885c9b51f82f3"]}` to the end of the url and it will I think take you to exactly the right place.